verify/verification.py

```diff
@@ -167,39 +306,87 @@ async def verify_agree(self, ctx):
             return False
 
         toggle = await self.config.guild(member.guild).toggle()
-        role_config = await self.config.guild(member.guild).role()
+        role_config = await self.config.guild(member.guild).temprole()
         if toggle is False:
             return False
 
         if role_config is not None:
             role = discord.utils.get(member.guild.roles, id=int(role_config))
             try:
-                await member.add_roles(role, reason="New Member, Needs to Verify")
+                await member.add_roles(role, reason="New Member, Needs to Verify.")
             except discord.Forbidden:
                 return self.log.warning("Unable to grant roles to new joiner.")
             except discord.HTTPException as e:
                 return self.log.warning("HTTPException: {} - {}".format(e.status, e.code))
+
+    async def _handle_role(self, member: discord.Member) -> tuple:
+        """Function to give and/or remove role from member.
+        This will automatically grab roles from server's config.
+
+        Parameters:
+            member: discord.Member, The member we will act on.
+
+        Returns:
+            In a tuple:
+            - bool: True if adding/remove role succeeded, else False.
+            - str: A string to use for logs purpose.
+
+        Raise:
+            discord.Forbidden: Missing permissions to do role handling.
+        """
+        list_to_add = await self.config.guild(member.guild).autoroles()
+        list_to_remove = await self.config.guild(member.guild).temprole()
+        actions = []
+        if list_to_add:
+            for role in list_to_add:
+                to_add = member.guild.get_role(role)
+                await member.add_roles(to_add, reason="Adding auto role by Verify.")
+            actions.append(
+                "added automatically role{plural}".format(
+                    plural="s" if len(list_to_add) > 1 else ""
+                )
+            )
+        if list_to_remove:
+            to_remove = member.guild.get_role(list_to_remove)
+            if to_remove in member.roles:
+                await member.remove_roles(to_remove, reason="Removing temporary role by Verify.")
+                actions.append("removed temporary role")
+        return (
+            True,
+            humanize_list(actions).capitalize() if actions else "No action taken.",
+        )
+
+    async def _maybe_update_config(self):
+        if not await self.config.version():  # We never had a version before
+            guild_dict = await self.config.all_guilds()
+            async for guild_id, info in AsyncIter(guild_dict.items()):
+                old_temporary_role = info.get("role", None)
+                if old_temporary_role:
+                    await self.config.guild_from_id(guild_id).temprole.set(old_temporary_role)
+                    await self.config.guild_from_id(guild_id).role.clear()
+            await self.config.version.set("1.0.0")
```

Comment:
Has been tested and seem to work as intended.
[#testing](https://canary.discord.com/channels/133049272517001216/133251234164375552/764165421670203413) to check tests.